src/client/Microsoft.Identity.Client/AuthenticationHeaderParser.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.PlatformsCommon.Factories;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Parsed authentication headers to retreve header values from HttpResponseHeaders.
+    /// </summary>
+    public class AuthenticationHeaderParser
+    {
+        /// <summary>
+        /// Parameters returned by the WWW-Authenticate header. This allows for dynamic
+        /// scenarios such as claim challenge, CAE, CA auth context.
+        /// See https://aka.ms/msal-net/wwwAuthenticate.
+        /// </summary>
+        public IReadOnlyList<WwwAuthenticateParameters> WwwAuthenticateParameters { get; private set; }
```

Comment:
This can be defaulted to an empty list?

Reply:
it set to empty lists by default in the ParseAuthenticationHeaders method.